utils/checkAuth.js

```diff
@@ -0,0 +1,22 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+dotenv.config();
+
+export default (req, res, next) => {
+    const token = (req.headers.authorization || '').replace(/Bearer\s?/, '');
```

Comment:
Instead of using a regular expression to extract the token from the authorization header, use the split method to split the header by a space and select the second element of the array:
```suggestion
    const token = req.headers.authorization?.split(' ')[1] || '';
```